From 3dd1bf328e6f222eb7b6a227598c24bd7f9ee8bb Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Tue, 26 Apr 2016 05:22:28 +0000
Subject: [PATCH 2/3] Fix size (bits->bytes)

---
 src/jtag/drivers/bbg-swd.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/jtag/drivers/bbg-swd.c b/src/jtag/drivers/bbg-swd.c
index 0363f45..f4fd20b 100644
--- a/src/jtag/drivers/bbg-swd.c
+++ b/src/jtag/drivers/bbg-swd.c
@@ -186,20 +186,20 @@ static int bbg_swd_switch_seq(enum swd_special_seq seq)
 	case LINE_RESET:
 		LOG_DEBUG("SWD line reset");
 		pru_data_ram[0] = CMD_SIG_GEN | (swd_seq_line_reset_len << 8);
-		memcpy (&pru_data_ram[1], swd_seq_line_reset, swd_seq_line_reset_len);
+		memcpy (&pru_data_ram[1], swd_seq_line_reset, (swd_seq_line_reset_len+7)/8);
 		pru_request_cmd(pru_data_ram);
 		break;
 	case JTAG_TO_SWD:
 		LOG_DEBUG("JTAG-to-SWD");
-		pru_data_ram[0] = CMD_SIG_GEN | (swd_seq_jtag_to_swd_len << 8);
-		memcpy (&pru_data_ram[1], swd_seq_jtag_to_swd, swd_seq_jtag_to_swd_len);
+		pru_data_ram[0] = CMD_SIG_GEN | ((swd_seq_jtag_to_swd_len)<< 8);
+		memcpy (&pru_data_ram[1], swd_seq_jtag_to_swd, (swd_seq_jtag_to_swd_len+7)/8);
 		pru_request_cmd(pru_data_ram);
 		bbg_swd_idle(8);
 		break;
 	case SWD_TO_JTAG:
-		LOG_DEBUG("JTAG-to-SWD");
+		LOG_DEBUG("SWD-to-JTAG");
 		pru_data_ram[0] = CMD_SIG_GEN | (swd_seq_swd_to_jtag_len << 8);
-		memcpy (&pru_data_ram[1], swd_seq_swd_to_jtag, swd_seq_swd_to_jtag_len);
+		memcpy (&pru_data_ram[1], swd_seq_swd_to_jtag, (swd_seq_swd_to_jtag_len+7)/8);
 		pru_request_cmd(pru_data_ram);
 		break;
 	default:
-- 
2.1.4

