From c368c7f22b67ba3ce5034d0f66911b5df4dc3861 Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Tue, 26 Apr 2016 07:14:55 +0000
Subject: [PATCH 3/3] Implement use of NRST

---
 src/jtag/drivers/bbg-swd.c | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/src/jtag/drivers/bbg-swd.c b/src/jtag/drivers/bbg-swd.c
index f4fd20b..8158f5e 100644
--- a/src/jtag/drivers/bbg-swd.c
+++ b/src/jtag/drivers/bbg-swd.c
@@ -104,11 +104,7 @@ static int bbg_swd_close(void)
 }
 
 
-static int bbg_swd_gpio_srst(int on)
-{
-	/* XXX: not yet implemented */
-	return ERROR_OK;
-}
+static void bbg_swd_gpio_srst(int on);
 
 static bool swd_mode;
 
@@ -125,9 +121,7 @@ static int bbg_swd_interface_init(void)
 
 	if (jtag_reset_config & RESET_CNCT_UNDER_SRST) {
 		if (jtag_reset_config & RESET_SRST_NO_GATING) {
-			retval = bbg_swd_gpio_srst(0);
-			if (retval != ERROR_OK)
-				return ERROR_FAIL;
+			bbg_swd_gpio_srst(0);
 			LOG_INFO("Connecting under reset");
 		}
 	}
@@ -178,6 +172,14 @@ static void bbg_swd_idle(int count)
 	pru_request_cmd(pru_data_ram);
 }
 
+static void bbg_swd_gpio_srst(int signal)
+{
+	pru_data_ram[0] = CMD_GPIO_OUT;
+	pru_data_ram[1] = (1 << 15);
+	pru_data_ram[2] = signal ? 1 : 0;
+	pru_request_cmd(pru_data_ram);
+}
+
 static int bbg_swd_switch_seq(enum swd_special_seq seq)
 {
 	LOG_DEBUG("bbg_swd_switch_seq");
-- 
2.1.4

